src/core/reaction_ensemble.cpp

```diff
@@ -717,9 +717,9 @@ int ReactionAlgorithm::create_particle(int desired_type) {
   // for components
   double vel[3];
   // we use mass=1 for all particles, think about adapting this
-  vel[0] = std::pow(2 * PI * temperature, -3.0 / 2.0) * gaussian_random();
-  vel[1] = std::pow(2 * PI * temperature, -3.0 / 2.0) * gaussian_random();
-  vel[2] = std::pow(2 * PI * temperature, -3.0 / 2.0) * gaussian_random();
+  vel[0] = std::sqrt(temperature) * gaussian_random();
```

Comment:
This should read  vel[0] = std::sqrt(temperature/particle_mass) * gaussian_random(), etc since the Maxwell-Boltzmann probability density is mass dependent. However particle_mass would be needed to be set in the reaction ensemble interface (in the same way the charge is set). Maybe we should also rename tempertature to kbT.

Reply:
I would leave it as the temperature, as written in the manual:

> A special note is due regarding the temperature, which is coupled to the energy scale by Boltzmann’s constant. However, since ESPResSo does not enforce a particular unit system, we also don’t know the numerical value of the Boltzmann constant in the current unit system. Therefore, when specifying the temperature of a thermostat, you actually do not define the temperature, but the value of the thermal energy kBT in the current unit system.

Therefore it is kind of implied that the temperature equals kbT.

Reply:
I added a comment on the missing mass in the documentation. I think that is enough until somebody wants to implement the correct mass.

Reply:
If we assign a new velocity to a particle in the global_mc move, do we have to take the change in the kinetic energy into account in the acceptance probability?

Reply:
To my understanding we do not need to take the kinetic energy into account in the acceptance probability. This would be in accordance with the algorithm described in https://aip.scitation.org/doi/pdf/10.1063/1.2359441?class=pdf (_Mesoscale simulation of polymer reaction equilibrium: Combining dissipative particle dynamics with reaction ensemble Monte Carlo. I. Polydispersed polymer systems_) [search for Maxwell-Boltzmann in this paper and also recognize that the kinetic energy does not occur in the acceptance probabilities]. The reasoning behind it is that drawing a Maxwell-Boltzmann distributed velocity from time to time resembles an Andersen Thermostat. The frequency with which a new velocity needs to be assigned is arbitrary (_Allen, Tildesley Computer Simulation of Liquids, 1991, p. 228_).

Reply:
I think you are right, as long as we sample from the correct distribution of the kinetic energy, then we do not need to take the kinetic energy into account. (This follows quite easily from the Hastings formalism...)

And apparently the Maxwell distribution is valid generally (see https://doi.org/10.4236/jamp.2018.63052 and https://chemistry.stackexchange.com/a/94294). I am not sure yet if this applies to more "exotic" systems (relativistic systems and quantized systems), but I guess these are not represented in espressomd anyways.

Still, I would propose, to address this in a future pull request. At least, I don't feel confident enough yet in this section of the code to claim that I know what I am doing.

Reply:
I think if you would add the lines 

```
double vel[3];
vel[0] = std::sqrt(temperature) * gaussian_random();
vel[1] = std::sqrt(temperature) * gaussian_random();
vel[2] = std::sqrt(temperature) * gaussian_random();
set_particle_v(p_id, vel);
```
below https://github.com/espressomd/espresso/blob/314115186f79749bda8bfc3280cb37dfa03fea48/src/core/reaction_ensemble.cpp#L836

this would fix the problem #1770 (at least for the case that we have particles with mass unity). We could then postpone the case for particles with different masses than 1 (and no other thermostat which does the correct thermalization).